ArjunaCore/arjuna/tests/classes/com/hp/mwtests/ts/arjuna/atomicaction/AtomicActionAsyncTest.java

```diff
@@ -49,6 +56,28 @@ public void testAbort() throws Exception {
     }
 
     @Test
+    public void testAbortAsyncAC() throws Exception {
+        AtomicActionTestBase.init(false);
+        arjPropertyManager.getCoordinatorEnvironmentBean().setAsyncAfterSynchronization(true);
+
+        final Thread thread = Thread.currentThread();
+        executeTest(true, ActionStatus.COMMITTED, new SyncRecord[] { new SyncRecord() {
+            @Override
+            public boolean afterCompletion(int status) {
+                testAbortAsyncAC = testAbortAsyncAC && (Thread.currentThread() != thread);
+                return true;
+            }
+        }, new SyncRecord(true, SyncRecord.FailureMode.NONE) {
+            @Override
+            public boolean afterCompletion(int status) {
+                testAbortAsyncAC = testAbortAsyncAC && (Thread.currentThread() != thread);
+                return true;
+            }
+        }}, new BasicRecord(), new BasicRecord());
+        assertTrue(testAbortAsyncAC);
```

Comment:
testAbortAsyncAC starts out true so if there is a bug which results in the synchronizations never getting called then this assert will be vacuously true.